test/insertStyle.test.ts

```diff
@@ -0,0 +1,41 @@
+import test from "ava";
+import insertStyle from "../src/insertStyle";
+import { Browser } from "happy-dom";
+
+const expectA = "body{color:red}";
+
+test("should insertStyle works", async (t) => {
+  const browser = new Browser();
+  const page = browser.newPage();
+
+  page.url = "https://example.com";
+  page.content = `<html><head></head><body></body></html>`;
+
+  // @ts-expect-error
+  global["window"] = page.mainFrame.window;
+  // @ts-expect-error
+  global["document"] = page.mainFrame.window.document;
+
+  const cssStr = insertStyle(expectA);
+
+  const styleSheet = document.head.querySelector("style")!;
+  t.is(
+    styleSheet.textContent,
+    cssStr!,
+    "stylesheet's content should equal returned css string"
+  );
+  t.is(
+    styleSheet.type,
+    "text/css",
+    'Should contain `type` attrib. equal to "text/css"'
+  );
+
+  await browser.close();
+});
+
+test("insertStyle shouldn't choke when window is undefined", (t) => {
+  delete global["window"];
+  t.throws(() => !window);
```

Comment:
`throw` test seems un-required here - reasoning?

Reply:
I can only guess the reason... this assertion is already present on the main branch
https://github.com/elycruz/rollup-plugin-sass/blob/0425739612e38ded0fda00e5060b25a735d620c3/test/style.test.ts#L31

the tests are probably executed in `strict` mode so an error occurs if you access a variable that doesn't exists